src/widgets/splits/SplitContainer.cpp

```diff
@@ -799,6 +804,54 @@ void SplitContainer::popup()
     window.show();
 }
 
+NodeDescriptor SplitContainer::buildDescriptorRecursively(Node *currentNode)
+{
+    if (currentNode->children_.empty())
+    {
+        const auto channelTypeToString = [](Channel::Type type) {
+            switch (type)
+            {
+                case Channel::Type::Twitch:
+                    return "twitch";
+                case Channel::Type::TwitchWhispers:
+                    return "whispers";
+                case Channel::Type::TwitchWatching:
+                    return "watching";
+                case Channel::Type::TwitchMentions:
+                    return "mentions";
+                case Channel::Type::TwitchLive:
+                    return "live";
+                case Channel::Type::TwitchAutomod:
+                    return "automod";
+                case Channel::Type::Irc:
+                    return "irc";
+                case Channel::Type::Misc:
+                    return "misc";
+                default:
+                    return "";
+            }
+        };
+
+        SplitNodeDescriptor result;
+        result.type_ =
+            channelTypeToString(currentNode->split_->getChannel()->getType());
```

Comment:
> Looks like you force pushed Pajs commits out of existence, can look for them upward from here https://github.com/Chatterino/chatterino2/commit/44f22f732559ab3065d5fcadb33028b3ad5faec8 :D

Oops, my alias did a bit of trolling. Didn't realize someone else pushed to the branch. Not sure if there is an easy way for me to recover it now that the commit is orphan

> descriptor building doesn't handle empty nodes rn, so trying to duplicate empty tab (Node::Type::EmptyRoot) will crash here because split is null. We probably should allow that for ux consistency

Makes sense, I'll add a check for that

Reply:
> Oops, my alias did a bit of trolling. Didn't realize someone else pushed to the branch. Not sure if there is an easy way for me to recover it now that the commit is orphan

If you know the SHA, you can append `.diff` or `.patch` to the commit's URL on GitHub - for example <https://github.com/Chatterino/chatterino2/commit/44f22f732559ab3065d5fcadb33028b3ad5faec8.patch>.

Reply:
I could get the changes from a patch and make a new commit out of that, is that ok?

Reply:
> If you know the SHA, you can append .diff or .patch to the commit's URL on GitHub

We commented basically at the same time lol. Yeah, I will do that and add a new commit with pajlada's changes.

Reply:
patches are fine, rebasing onto 44f22f732559ab3065d5fcadb33028b3ad5faec8 should also work or cherry picking skipped commits 
```
git cherry-pick 2f4021bfd9c1b5e156942e3bc8b880524fb5e96c
git cherry-pick 75027efb07517b70a1a535de97e1f9c6408b17e3
git cherry-pick bfce89c94d9858b2bf4fc53c63971491d4a842e7
git cherry-pick 470cd2ae33114cbb7e83a9996653500e42b484c2
git cherry-pick 0184bc49446f2f6b5f159c268c7e7973eb2b45f0
```

Reply:
I wasn't able to find the commits by fetching (I guess because they aren't on any branch) so `cherry-pick` wasn't working

```
 $ git cherry-pick 470cd2ae33114cbb7e83a9996653500e42b484c2       
fatal: bad object 470cd2ae33114cbb7e83a9996653500e42b484c2
```